### DISCUSSION TITLE
CompoundBody collide with each other and produce unexpected wobbles

### DISCUSSION THREAD

**Discussion by c068162** [bepu/bepuphysics1#51] [Q&A]
hello.
I'm using BepuV1 to make a game similar to bumper cars with a TimeStepDuration of 120 and a logical Tick of 24, and everything looks perfect when I use a single Box as a rigid body because of the damage determination of the parts, So I used CompoundBody and CompoundChildData, but when two Compoundbodies collided with each other through ApplyImpulse, they would jitter unexpectedly and even penetrate each other, But I use the PositionUpdateMode. Continuous, they are in contact with will produce unexpected jitter, hope you can give me some advice.
thank you.

**Comment by RossNordby** [comment]
Sorry, there's not quite enough information to pin down the source if the issue. I assume by a TimeStepDuration of 120 you mean 1/120f; if it is actually 120, that is very likely causing problems.

Assuming it's not related to that, you could try increasing the solver's iteration count to see if that stabilizes the simulation.

It may also have something to do with how the impulse is applied, the magnitude of the impulse, or how often the impulse is applied. Impulses are instantaneous changes in momentum; if the changes are very large, they could overpower the default contact constraint softness or it difficult to solve. A reproduction of the issue in the demos would help narrow it down.

**Reply by c068162** [reply]
Thank you for your reply
Yes, I use a TimeStepDuration of 1/120f, I modify solver's iteration count, but it doesn't seem to change much, all my moves and sprints are done by ApplyImpulse per frame, which may have something to do with it, I upload a short video, The first impact produced a shake.

https://github.com/bepu/bepuphysics1/assets/121477712/7d42c91d-b3ad-4d8b-9a6b-0ea1f778632f

**Reply by RossNordby** [reply]
That's very odd behavior for the simulation in isolation; it looks like interacting bodies in a multiplayer environment running into local disagreements in simulation result that then get corrected. If that is the case, I'd recommend running a similar test without any networking involved to confirm the behavior still persists.

If that is all fully local simulation with no networking involved at all, I'll probably need a reproduction in the demos; it's too strange to diagnose otherwise.

**Reply by c068162** [reply]
I think I found the problem, because recently I changed the smoothing scheme, the local player smoothing works fine, the remote player smoothing will shake.
I'm sorry
